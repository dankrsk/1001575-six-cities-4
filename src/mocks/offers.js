const TYPES = [
  `Apartment`,
  `Room`,
  `House`,
  `Hotel`,
];

const TITLES = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
  `Wood and stone place`,
];

const getOffer = () => {
  return {
    id: Math.round(Math.random() * 10000),
    type: TYPES[Math.round(Math.random() * 3)],
    title: TITLES[Math.round(Math.random() * 4)],
    price: Math.round(Math.random() * 100),
    rating: Math.round(Math.random() * 5),
    isPremium: Math.random() > 0.5 ? true : false,
    photo: `http://cdn.home-designing.com/wp-content/uploads/2013/11/1-Contemporary-architecture.jpg`,
  };
};

export const getOffers = (count) => {
  return Array(count).fill(``).map(() => getOffer());
};
