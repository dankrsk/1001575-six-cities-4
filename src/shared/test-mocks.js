export const mock = {
  rentOffersCount: 3,
  offers: [
    {
      id: 1,
      type: `House`,
      title: `Beautiful & luxurious apartment at great location`,
      price: 100,
      rating: 1,
      isPremium: true,
      photo: `http://cdn.home-designing.com/wp-content/uploads/2013/11/1-Contemporary-architecture.jpg`,
      coordinates: [52.3909553943508, 4.929309666406198],
    },
    {
      id: 2,
      type: `Room`,
      title: `Wood and stone place`,
      price: 200,
      rating: 2,
      isPremium: false,
      photo: `https://avatars.mds.yandex.net/get-pdb/215709/4d8ee911-7746-43da-a5cd-b7f9d2f3f839/s1200?webp=false`,
      coordinates: [52.369553943508, 4.85309666406198],
    },
    {
      id: 3,
      type: `Appartment`,
      title: `Nice, cozy, warm big bed apartment`,
      price: 300,
      rating: 3,
      isPremium: true,
      photo: `https://avatars.mds.yandex.net/get-pdb/215709/4d8ee911-7746-43da-a5cd-b7f9d2f3f839/s1200?webp=false`,
      coordinates: [52.3909553943508, 4.85309666406198],
    },
  ]
};
