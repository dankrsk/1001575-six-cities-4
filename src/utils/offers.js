const CITIES_COUNT = 6;

export const getOffersByCity = (city, offers) => {
  return offers.filter((offer) => {
    return offer.city === city;
  });
};

export const getCitiesFromOffers = (offers) => {
  const cities = [];
  let i = 0;

  while (cities.length < CITIES_COUNT && i < offers.length) {
    const city = offers[i].city;

    if (!cities.includes(city)) {
      cities.push(city);
    }

    i++;
  }

  return cities;
};
