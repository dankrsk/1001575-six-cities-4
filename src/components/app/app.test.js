import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import App from '../app/app.jsx';
import configureStore from 'redux-mock-store';
import {mock} from '../../shared/test-mocks.js';
import {getCitiesFromOffers} from '../../utils/offers.js';
import NameSpace from '../../const.js';

const mockStore = configureStore([]);

describe(`Snapshots for App`, () => {
  it(`App component`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        offers: mock.offers,
        status: `OK`,
      },
      [NameSpace.APP]: {
        city: `Paris`,
        allCities: getCitiesFromOffers(mock.offers),
      },
      [NameSpace.USER]: {
        authorizationStatus: `NO_AUTH`,
      },
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <App />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
