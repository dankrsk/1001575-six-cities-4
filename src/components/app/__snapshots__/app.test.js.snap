// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots for App App component 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign In
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
              onClick={[Function]}
            >
              <span>
                Amsterdam
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
              onClick={[Function]}
            >
              <span>
                Cologne
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
              onClick={[Function]}
            >
              <span>
                Berlin
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            1
             places to stay in 
            Amsterdam
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by 
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              onKeyDown={[Function]}
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom"
            >
              <li
                className="places__option places__option--active"
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="cities"
          >
            <div
              className="places__list near-places__list"
            >
              <article
                className="place-card cities__place-card"
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="place-card__image-wrapper cities__image-wrapper"
                >
                  <a>
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="200"
                      src="http://cdn.home-designing.com/wp-content/uploads/2013/11/1-Contemporary-architecture.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  className="place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        100
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "20%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/1"
                      onClick={[Function]}
                    >
                      Beautiful & luxurious apartment at great location
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    House
                  </p>
                </div>
              </article>
            </div>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
